text/0000-multiarch-builders-and-package.md

```diff
@@ -0,0 +1,944 @@
+```
+
+In these cases each `target` entry corresponds to a different buildpack image that is exported into an [image index](https://github.com/opencontainers/image-spec/blob/master/image-index.md)
+
+### Package a multi-arch Builder
+
+In case of packing a **Builder**, we assume the following premises: 
+
+1. Buildpack authors migrated their `builder.toml` and [remove the stack](https://github.com/buildpacks/pack/issues/1303) concept.
+2. Multi-architecture `build`, `run` images and `buildpacks` are available for baking into the **Builder**.
+
+A sample `builder.toml` file looks like:
+
+```toml
+# Buildpacks to include in builder, these buildpacks MUST be multi-arch and point to Image Index
+[[buildpacks]]
+uri = "<some uri>"
+
+[run]
+# Runtime images - in case of multi-arch images it must point to Image Index
+[[run.images]]
+image = "index.docker.io/paketobuildpacks/run-jammy-tiny:latest"
+
+[build]
+# This image is used at build-time, in case of multi-arch images it must point to Image Index
+image = "docker.io/paketobuildpacks/build-jammy-tiny:0.2.3"
+
+[[targets]]
+os = "linux"
+arch = "amd64"
+
+[[targets]]
+os = "linux"
+arch = "arm64"
+```
+
+As we can see, the proposal is based on the assumption that the `run-image`, `build-image` and `buildpacks` to include 
+in the builder are **multi-arch artifacts**, and we can reach them by reading an 
+[image index](https://github.com/opencontainers/image-spec/blob/master/image-index.md)
+
+```bash
+pack builder create my-jammy-builder --config ./builder.toml \ 
+      --multi-arch \ 
+      --publish
+```
+
+In this case, because `targets` are specified and `--multi-arch` is being used `pack` will follow the builder creation process for **each provided platform**, 
+pulling the correct (based on os/arch) buildpacks, build and run images and creating different builders images that are 
+exported and combined into an [image index](https://github.com/opencontainers/image-spec/blob/master/image-index.md)
+
+# How it Works
+[how-it-works]: #how-it-works
+
+## Buildpack Package
+
+As a quick summary, our current process to create a buildpack package involves:
+
+- The end-users defined `os` for the OCI image using the [package.toml](https://buildpacks.io/docs/reference/config/package-config/) when they want to export a buildpack package.
+- The only values allowed are `linux` and `windows` and by default when is not present, `linux` is being used.
+- When exporting to daemon, the `docker.OSType` must be equal to `platform.os`
+
+### To keep compatibility 
+
+We propose:
+- Deprecate the `platform.os` field from [package.toml](https://buildpacks.io/docs/reference/config/package-config/). It will be removed after two pack releases with the new feature
+- When `platform.os` is present in [package.toml](https://buildpacks.io/docs/reference/config/package-config/), throw a warning messages indicating the field will be removed 
+and `--platform` flag must be used
```

Comment:
This was brought up elsewhere, but `--platform` might be confusing when the corresponding word in buildpack.toml and builder.toml is `--target`.